client/blocks/comments/comment-actions.jsx

```diff
@@ -0,0 +1,123 @@
+import { changeCommentStatus } from 'state/comments/actions';
+import CommentLikeButtonContainer from './comment-likes';
+import CommentApproveAction from './comment-approve-action';
+import EllipsisMenu from 'components/ellipsis-menu';
+import PopoverMenuItem from 'components/popover/menu-item';
+import PopoverMenuSeparator from 'components/popover/menu-separator';
+
+const CommentActions = ( {
+	post,
+	comment: { isPlaceholder, status },
+	translate,
+	activeEditCommentId,
+	activeReplyCommentID,
+	commentId,
+	handleReply,
+	onReplyCancel,
+	approveComment,
+	unapproveComment,
+	trashComment,
+	spamComment,
+	editComment,
+	editCommentCancel,
+ } ) => {
+	const showReplyButton = post && post.discussion && post.discussion.comments_open === true;
+	const showCancelReplyButton = activeReplyCommentID === commentId;
+	const showCancelEditButton = activeEditCommentId === commentId;
+	const isApproved = status === 'approved';
+
+	// Only render actions for non placeholders
+	if ( isPlaceholder ) {
+		return null;
+	}
+
+	return (
+		<div className="comments__comment-actions">
+			{ showReplyButton &&
+				<button className="comments__comment-actions-reply" onClick={ handleReply }>
+					<Gridicon icon="reply" size={ 18 } />
+					<span className="comments__comment-actions-reply-label">{ translate( 'Reply' ) }</span>
+				</button>
+			}
+			{ showCancelReplyButton &&
+				<button className="comments__comment-actions-cancel-reply" onClick={ onReplyCancel }>{ translate( 'Cancel reply' ) }</button>
+			}
+			{ showCancelEditButton &&
+				<button className="comments__comment-actions-cancel-reply" onClick={ editCommentCancel }>{ translate( 'Cancel' ) }</button>
+			}
+			<CommentLikeButtonContainer
+				className="comments__comment-actions-like"
+				tagName="button"
+				siteId={ post.site_ID }
+				postId={ post.ID }
+				commentId={ commentId }
+			/>
+			{ isEnabled( 'comments/moderation-tools-in-posts' ) &&
+				<CommentApproveAction { ...{ status, approveComment, unapproveComment } } />
+			}
+			{ isEnabled( 'comments/moderation-tools-in-posts' ) &&
+				<button className="comments__comment-actions-trash" onClick={ trashComment }>
+					<Gridicon icon="trash" size={ 18 } />
+					<span className="comments__comment-actions-like-label">{ translate( 'Trash' ) }</span>
+				</button>
+			}
+			{ isEnabled( 'comments/moderation-tools-in-posts' ) &&
+				<button className="comments__comment-actions-spam" onClick={ spamComment }>
+					<Gridicon icon="spam" size={ 18 } />
+					<span className="comments__comment-actions-like-label">{ translate( 'Spam' ) }</span>
+				</button>
+			}
+			{ isEnabled( 'comments/moderation-tools-in-posts' ) &&
+				<button className="comments__comment-actions-edit" onClick={ editComment }>
+					<Gridicon icon="pencil" size={ 18 } />
+					<span className="comments__comment-actions-like-label">{ translate( 'Edit' ) }</span>
+				</button>
+			}
+			{ isEnabled( 'comments/moderation-tools-in-posts' ) &&
```

Comment:
I think you'll need to add some checks for the user type. Certain WordPress user roles can't moderate comments, so we'd want to hide the moderate actions for them.

https://codex.wordpress.org/Roles_and_Capabilities